src/main/java/space/obminyashka/items_exchange/service/impl/UserServiceImpl.java

```diff
@@ -76,6 +76,7 @@ public boolean registerNewUser(UserRegistrationDto userRegistrationDto) {
     public User loginUserWithOAuth2(DefaultOidcUser oauth2User) {
         var optionalUser = findByUsernameOrEmail(oauth2User.getEmail());
         optionalUser.ifPresent(ignored -> setOAuth2LoginToUserByEmail(oauth2User.getEmail()));
+        optionalUser.ifPresent(this::setValidatedEmailToUserLoggedWithOAuth2ByEmail);
```

Comment:
You can do the same without a new method simply by using Optional's features and making that in a functional way:
```
        optionalUser.filter(User::isValidatedEmail)
                .map(User::getEmail)
                .ifPresent(userRepository::setValidatedEmailToUserLoggedWithOAuth2ByEmail);
```